File: Sources/ChargeCat/Battery/BatterySnapshot.swift
import Foundation

struct BatterySnapshot: Equatable {
    let level: Int
    let isPluggedIn: Bool
    let isCharging: Bool

    var powerText: String {
        if isPluggedIn && level >= 99 && isCharging == false {
            return "Fully Charged"
        }
        if isCharging {
            return "Charging"
        }
        if isPluggedIn {
            return "Power Connected"
        }
        return "On Battery"
    }
}

File: windows/ChargeCat.Windows/Battery/BatterySnapshot.cs
namespace ChargeCat.WindowsApp.Battery;

internal sealed record BatterySnapshot(int Level, bool IsPluggedIn, bool IsCharging)
{
    public string PowerText =>
        IsPluggedIn && Level >= 99 && IsCharging == false ? "Fully Charged" :
        IsCharging ? "Charging" :
        IsPluggedIn ? "Power Connected" :
        "On Battery";
}
